internal/prototest/util.go

```diff
@@ -75,174 +74,15 @@ func findFileInSet(fps *descriptorpb.FileDescriptorSet, name string) *descriptor
 	return nil
 }
 
-func checkFileDescriptor(t *testing.T, act, exp *descriptorpb.FileDescriptorProto) {
-	compareFiles(t, fmt.Sprintf("%q", act.GetName()), exp, act)
-}
-
-// adapted from implementation of proto.Equal, but records an error for each discrepancy
-// found (does NOT exit early when a discrepancy is found).
-func compareFiles(t *testing.T, path string, exp, act *descriptorpb.FileDescriptorProto) {
-	if (exp == nil) != (act == nil) {
-		if exp == nil {
-			t.Errorf("%s: expected is nil; actual is not", path)
-		} else {
-			t.Errorf("%s: expected is not nil, but actual is", path)
-		}
-		return
-	}
-	mexp := exp.ProtoReflect()
-	mact := act.ProtoReflect()
-	if mexp.IsValid() != mact.IsValid() {
-		if mexp.IsValid() {
-			t.Errorf("%s: expected is valid; actual is not", path)
-		} else {
-			t.Errorf("%s: expected is not valid, but actual is", path)
-		}
-		return
-	}
-	CompareMessages(t, path, mexp, mact)
-}
-
-func CompareMessages(t *testing.T, path string, exp, act protoreflect.Message) {
+func CompareMessages(t *testing.T, path string, exp, act protoreflect.Message, opts ...cmp.Option) {
```

Comment:
Not using this yet, but added variable arguments to pass options to comparison (if we ever want to filter out anything from the comparison).

Reply:
I think you can get rid of the `path` argument and then use this new function from up above on line 56.

Maybe rename `AssertMessagesEqual`?

Reply:
The `path` argument is passed via another method in `source_code_info_test.go` so we can't remove it. Also if we replace the usage in `checkFiles` we'd have to convert back from proto.Message to protoreflect.Message.

Reply:
Okay, my bad. I misunderstood when you said "Not using this yet". I thought maybe you were referring to entire function.

For the way this is used, I think maybe something like this would be better:
```
    AssertMessagesEqual(t, expected, actual, "%s: messages should match", path)
```
That way it's more like other assertion functions and less bespoke to the current usage from `source_code_info_test.go`.

> we'd have to convert back from proto.Message to protoreflect.Message

Oy, I didn't even notice the difference in types. I think `proto.Message` is likely the more commonly used type, so that makes the most sense on a helper method (future uses would be more likely to use that type). It seems like not a big deal to move the invocation of `.Interface()` out of this helper and to the call site in `source_code_info_test.go`. Then you can re-use from both places.